config/crd/bases/atlas.mongodb.com_atlasdatabaseusers.yaml

```diff
@@ -54,6 +54,20 @@ spec:
             description: AtlasDatabaseUserSpec defines the desired state of Database
               User in Atlas
             properties:
+              atlasRef:
+                description: AtlasRef holds the Atlas project ID the user belongs
+                  to
+                properties:
+                  credentials:
+                    description: Credentials is the name of the secret that holds
```

Comment:
Are we assume credentials will always be on the same namespace as the user? I think it should be a `ResourceRefNamespaced` (just like connection secret) type according to the SPIKE

Reply:
I disagree, it does not seem sensible security wise to be grabbing credentials from some other namespace. Specially in this case that credentials are per CRD they can be locally provisioned.

Reply:
Actually, I have reconsidered this while talking to @helderjs.

Seems the general recommendation is that all references should be without the namespace, with a single exception being the secrets themselves.

This is because, normal resources should not depend or reach out to other namespaces, as that would have security concerns and undermines the actual reason for having namespaces in the first place. Still, in the case of secrets, it makes sense to actually allow users of a namespace to consume secrets even when they are not allowed to create or update them themselves.

So I guess yes, we should also allow namespaces for these credentials here.

Reply:
> it does not seem sensible security wise to be grabbing credentials from some other namespace.

agreed

Reply:
Reconsidering this:
> Still, in the case of secrets, it makes sense to actually allow users of a namespace to consume secrets even when they are not allowed to create or update them themselves.

[Implementing Anti-patterns: Kubernetes Cross-namespace Resource Ownership - Tom Coufal, Red Hat](https://www.youtube.com/watch?v=iWz5AAbbT-c)

We still can be Kubernetes best practices compliant and allow for managers of secrets and consumers to be different:
- Just use a namespaced used for admins with an RBAC to create or update secrets in the namespace.
- Use a different namespaced user withan RBAC for regular devs to just consume the secrets in the namespace.
- Should a secret be needed in several namespaces? just provision it on all of them. Still, if it needs to be in different namespaces maybe it should NOT be the same exact credential, it might help security to use different credentials for different dev teams or end users.

Reply:
Resolved as won't do